dbcon/joblist/lbidlist.cpp

```diff
@@ -653,32 +680,46 @@ inline bool LBIDList::compareVal(const T& Min, const T& Max, const T& value, cha
     return true;
 }
 
-bool LBIDList::checkSingleValue(int64_t min, int64_t max, int64_t value,
+template<typename T>
+bool LBIDList::checkSingleValue(T min, T max, T value,
                                 execplan::CalpontSystemCatalog::ColDataType type)
 {
     if (isCharType(type))
     {
+        // MCOL-641 LBIDList::CasualPartitionDataType() returns false if
+        // width > 8 for a character type, so T cannot be __int128 here
         uint64_t mmin = order_swap(min);
         uint64_t mmax = order_swap(max);
         uint64_t vvalue = order_swap(value);
         return (vvalue >= mmin && vvalue <= mmax);
     }
     else if (isUnsigned(type))
     {
-        return (static_cast<uint64_t>(value) >= static_cast<uint64_t>(min) &&
-                static_cast<uint64_t>(value) <= static_cast<uint64_t>(max));
+        if (typeid(T) == typeid(__int128))
+        {
+            return (static_cast<unsigned __int128>(value) >= static_cast<unsigned __int128>(min) &&
+                    static_cast<unsigned __int128>(value) <= static_cast<unsigned __int128>(max));
+        }
+        else
+        {
+            return (static_cast<uint64_t>(value) >= static_cast<uint64_t>(min) &&
+                    static_cast<uint64_t>(value) <= static_cast<uint64_t>(max));
+        }
     }
     else
     {
         return (value >= min && value <= max);
```

Comment:
Will we compare wide-DECIMALs getting into else branch?

Reply:
I think the first question is: if it is possible to get here with wide-DECIMALs in the first place.

Reply:
Yes. The calling function, TupleBPS::addCPPredicates(), filters out extents of the large-side table, based on the CP info of the small-side tables. So if either the small-side or the large-side table contains wide-decimal columns as join keys (which we currently do not support), we will get here for wide-decimals.

Reply:
And I c that we don't treat value based on the dt used. Should't we have the same cimparison as in preceding blocks?

Reply:
Not sure I understand the last comment. The preceding else block is for unsigned types, for which the comparison involves first casting to the unsigned equivalent of either in128_t or int64_t. For the last else condition (signed datatypes), we compare the values directly.